source/qb64.bas

```diff
@@ -25116,17 +25126,18 @@ END FUNCTION
 
 
 FUNCTION QuotedFilename$ (f$)
+    SELECT CASE os$
+        CASE "WIN": QuotedFilename$ = CHR$(34) + f$ + CHR$(34)
 
-    IF os$ = "WIN" THEN
-        QuotedFilename$ = CHR$(34) + f$ + CHR$(34)
-        EXIT FUNCTION
-    END IF
-
-    IF os$ = "LNX" THEN
-        QuotedFilename$ = "'" + f$ + "'"
-        EXIT FUNCTION
-    END IF
+        CASE "LNX"
+            i~% = INSTR(f$, "'") 'do escapes to be safe
+            WHILE i~%
+                f$ = LEFT$(f$, i~%) + "\" + MID$(f$, i~%) 'escape "'" (' -> \')
+                i~% = INSTR(i~% + 2, f$, "'")
+            WEND

```

Comment:
This function is not correct, it returns a string mostly full of single quotes.

On top of that, the intended behaviour is not correct either. A single-quote cannot occur with single-quotes because backslash escapes don't work in single-quotes. Instead you need to close the quoted string, add the single-quote, then open the single-quoting again. Example of this function's desired  input and output:
`the'rain'in` -> `'the'\''rain'\''in'`